packages/dds/matrix/src/matrix.ts

```diff
@@ -788,36 +772,43 @@ export class SharedMatrix<T = any>
 				this.rows.applyMsg(msg, local);
 				break;
 			default: {
-				if (contents.type === MatrixOp.changeSetCellPolicy) {
-					this.setCellLwwToFwwPolicySwitchOpSeqNumber = rawMessage.sequenceNumber;
-					return;
-				}
-
 				assert(
 					contents.type === MatrixOp.set,
 					0x021 /* "SharedMatrix message contents have unexpected type!" */,
 				);
 
-				const { row, col, value } = contents;
+				const { row, col, value, fwwMode } = contents;
+				const isPreviousSetCellPolicyModeFWW =
+					this.setCellLwwToFwwPolicySwitchOpSeqNumber > -1;
+				// If this is the first op notifying us of the policy change, then set the policy change seq number.
+				if (this.setCellLwwToFwwPolicySwitchOpSeqNumber === -1 && fwwMode === true) {
+					this.setCellLwwToFwwPolicySwitchOpSeqNumber = rawMessage.sequenceNumber;
+				}
+
 				assert(rawMessage.clientId !== null, "clientId should not be null!!");
 				if (local) {
 					// We are receiving the ACK for a local pending set operation.
 					const { rowHandle, colHandle, localSeq } = localOpMetadata as ISetOpMetadata;
-					if (this.isSetCellOpMadeInFWWPolicy(rawMessage)) {
-						if (this.shouldSetCellBasedOnFWW(rowHandle, colHandle, rawMessage)) {
-							this.cellLastWriteTracker.setCell(rowHandle, colHandle, {
-								seqNum: rawMessage.sequenceNumber,
-								clientId: rawMessage.clientId,
-							});
-						}
-					} else {
-						if (this.isLatestPendingWrite(rowHandle, colHandle, localSeq)) {
-							this.pending.setCell(rowHandle, colHandle, undefined);
-							this.cellLastWriteTracker.setCell(rowHandle, colHandle, {
-								seqNum: rawMessage.sequenceNumber,
-								clientId: rawMessage.clientId,
-							});
-						}
+					const isLatestPendingOp = this.isLatestPendingWrite(
+						rowHandle,
+						colHandle,
+						localSeq,
+					);
+					// If policy is switched and cell should be modified too based on policy, then update the tracker.
+					// If policy is not switched, then also update the tracker in case it is the latest.
+					if (
+						(this.setCellLwwToFwwPolicySwitchOpSeqNumber > -1 &&
+							this.shouldSetCellBasedOnFWW(rowHandle, colHandle, rawMessage)) ||
```

Comment:
So shouldSetCellBasedOnFWW() answers the question "is this FWW op a winner or loser case?", right? Maybe change comment in function from "should be applied" to "should resolve in conflict or be applied"

Reply:
Added assert, but no need to change comment as it will not necessarily result in conflict event, it depends whether the client had local changes or not. If not, then the op will just be dropped without doing anything.